pkg/trace/config/apply.go

```diff
@@ -247,17 +228,16 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		c.MaxMemory = config.Datadog.GetFloat64("apm_config.max_memory")
 	}
 
-	// undocumented
-	c.ServiceWriterConfig = readServiceWriterConfigYaml()
-	c.StatsWriterConfig = readStatsWriterConfigYaml()
-	c.TraceWriterConfig = readTraceWriterConfigYaml()
-
-	// allow 1% of maximum connnections for concurrent stats flushes
-	conns1percent := int(math.Max(1, float64(c.ConnectionLimit)/100))
-	c.StatsWriterConfig.SenderConfig.MaxConnections = conns1percent
-	// allow 10% of maximum connnections for concurrent trace flushes
-	conns10percent := int(math.Max(1, float64(c.ConnectionLimit)/10))
-	c.TraceWriterConfig.SenderConfig.MaxConnections = conns10percent
+	// undocumented writers
+	for key, cfg := range map[string]*SenderConfig{
+		"apm_config.service_writer": c.ServiceWriter,
+		"apm_config.trace_writer":   c.TraceWriter,
+		"apm_config.stats_writer":   c.StatsWriter,
```

Comment:
Since you've already answered your question, I'm not sure what more to say. I thought it would be confusing to set the defaults here, although in a way it makes sense. But then it also makes sense for the component itself to cover for 0-values. So I guess we're good?

Reply:
I'm still curious about the second part of the question:

> does this way of loading config support partial configuration (say, just configuring `QueueSize` and not `ConnectionLimit` which should arguable be part of the test suite btw)?

Reply:
Yes, of course. Sorry for missing that part. It does. I've already worked with this manually. It just unmarshal's whatever it finds, leaving everything else as zero values.